data-prepper-plugins/anomaly-detector-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/anomalydetector/AnomalyDetectorProcessor.java

```diff
@@ -45,7 +51,21 @@ private AnomalyDetectorMode loadAnomalyDetectorMode(final PluginFactory pluginFa
 
     @Override
     public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
-        return mode.handleEvents(records);
+        final List<Record<Event>> recordsOut = new LinkedList<>();
+
+        for (final Record<Event> record : records) {
+            final Event event = record.getData();
+            // If user has not configured IdentificationKeys, the empty set will always hash to "31",
+            // so the same forest will be used, and we don't need to write a special case.
+            final AggregateIdentificationKeysHasher.IdentificationKeysMap identificationKeysMap = aggregateIdentificationKeysHasher.createIdentificationKeysMapFromEvent(event);
+            if (!forestMap.containsKey(identificationKeysMap.hashCode())) {
```

Comment:
If identificationKeys are null, the code can be just 
```
return mode.handleEvents(records);
```

The lines 61-66 may be re-written like this?
```
forest = forestMap.get(identificationKeysMap.hashCode());
if (Objects.isNull(forestMode)) {
    forest = loadAnomalyDetectorMode(pluginFactory);
    forest.initialize(keys);
    forestMap.put(identificationKeysMap.hashCode(), forest)
}
recordsOut.addAll(forest.handleEvent(List.of(record)));
```